Matchismo/Matchismo/Controller/PlayingCardGameViewController.mm

```diff
@@ -26,116 +27,75 @@ - (Deck *)createDeck {
-                               regularExpressionWithPattern:spadesAndClubsPattern
-                                                    options:0
-                                                      error:nil];
-
-  auto textRange = NSMakeRange(0, moveDescription.length);
-  auto heartsAndDiamondsMatches = [heartsAndDiamondsRegex matchesInString:moveDescription
-                                                                  options:0
-                                                                    range:textRange];
-  auto spadesAndClubsMatches = [spadesAndClubsRegex matchesInString:moveDescription
-                                                            options:0
-                                                              range:textRange];
-  
-  auto *redColorAttribute = @{NSForegroundColorAttributeName : UIColor.redColor};
-  auto *blackColorAttribute = @{NSForegroundColorAttributeName : UIColor.blackColor};
-  auto attributedDescription = [[NSMutableAttributedString alloc] initWithString:moveDescription];
-  
-  for (NSTextCheckingResult *res in heartsAndDiamondsMatches) {
-    [attributedDescription addAttributes:redColorAttribute range:res.range];
-  }
-  
-  for (NSTextCheckingResult *res in spadesAndClubsMatches) {
-    [attributedDescription addAttributes:blackColorAttribute range:res.range];
+- (CardView *)createCardViewWithFrame:(CGRect)frame withCard:(Card *)card {
+  if (![card isKindOfClass:[PlayingCard class]]) {
+    return [[CardView alloc] initWithFrame:frame];
   }
+  PlayingCard *playingCard = (PlayingCard *)card;
+  auto playingCardView = [[PlayingCardView alloc] initWithFrame:frame];
+  playingCardView.suit = playingCard.suit;
+  playingCardView.rank = playingCard.rank;
+  playingCardView.faceUp = NO;
   
-  self.descriptionLabel.attributedText = attributedDescription;
+  return playingCardView;
 }
 
 - (void)updateUI {
-  for (UIButton *cardButton in self.cardsCollection) {
-    auto cardButtonIndex = [self.cardsCollection indexOfObject:cardButton];
-    Card *card = [self.game cardAtIndex:cardButtonIndex];
-    if ([card isKindOfClass:[PlayingCard class]]) {
-      auto playingCard = (PlayingCard *)card;
-      auto attributedTitle = [[NSAttributedString alloc]
-                                initWithString:[self titleForCard:card]
-                                    attributes:@{NSForegroundColorAttributeName
-                                                  : [self colorForCard:playingCard]}];
-      [cardButton setAttributedTitle:attributedTitle forState:UIControlStateNormal];
+  int i = 0;
+  while (i < self.cardViews.count) {
+    if (![self.cardViews[i] isKindOfClass:[PlayingCardView class]]) {
+      continue;
     }
-    [cardButton setBackgroundImage:[self backgroundImageForCard:card]
-                          forState:UIControlStateNormal];
-    [cardButton setEnabled:!card.matched];
+    PlayingCardView *playingCardView = (PlayingCardView *)self.cardViews[i];
```

Comment:
Don't worry about re-jigging the whole design or anything, but FYI it's generally a bit of a code smell if you have to do conditional logic with isKindOfClass + casting like this, so better to avoid designs like this if poss.